test/test-server/networking_handlers.go

```diff
@@ -1159,55 +1204,81 @@ func handleNetworkingPrivateLinkAttachmentConnectionGet(t *testing.T, id string)
 		Spec: &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionSpec{
 			PrivateLinkAttachment: &networkingprivatelinkv1.ObjectReference{Id: "platt-111111"},
-			Cloud: &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionSpecCloudOneOf{
-				NetworkingV1AwsPrivateLinkAttachmentConnection: &networkingprivatelinkv1.NetworkingV1AwsPrivateLinkAttachmentConnection{
-					Kind:          "AwsPrivateLinkAttachmentConnection",
-					VpcEndpointId: "vpce-01234567890abcdef",
-				},
-			},
-			DisplayName: networkingprivatelinkv1.PtrString(name),
-			Environment: &networkingprivatelinkv1.ObjectReference{Id: "env-00000"},
+			DisplayName:           networkingprivatelinkv1.PtrString(name),
+			Environment:           &networkingprivatelinkv1.ObjectReference{Id: "env-00000"},
 		},
 		Status: &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionStatus{
 			Phase: phase,
-			Cloud: &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionStatusCloudOneOf{
-				NetworkingV1AwsPrivateLinkAttachmentConnectionStatus: &networkingprivatelinkv1.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus{
-					Kind:                   "AwsPrivateLinkAttachmentConnectionStatus",
-					VpcEndpointId:          "vpce-01234567890abcdef",
-					VpcEndpointServiceName: "",
-				},
-			},
 		},
 	}
 
+	switch cloud {
+	case "aws":
+		connection.Spec.Cloud = &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionSpecCloudOneOf{
+			NetworkingV1AwsPrivateLinkAttachmentConnection: &networkingprivatelinkv1.NetworkingV1AwsPrivateLinkAttachmentConnection{
+				Kind:          "AwsPrivateLinkAttachmentConnection",
+				VpcEndpointId: "vpce-01234567890abcdef",
+			},
+		}
+		connection.Status.Cloud = &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionStatusCloudOneOf{
+			NetworkingV1AwsPrivateLinkAttachmentConnectionStatus: &networkingprivatelinkv1.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus{
+				Kind:                   "AwsPrivateLinkAttachmentConnectionStatus",
+				VpcEndpointId:          "vpce-01234567890abcdef",
+				VpcEndpointServiceName: "",
+			},
+		}
+	case "azure":
+		connection.Spec.Cloud = &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionSpecCloudOneOf{
+			NetworkingV1AzurePrivateLinkAttachmentConnection: &networkingprivatelinkv1.NetworkingV1AzurePrivateLinkAttachmentConnection{
+				Kind:                      "AzurePrivateLinkAttachmentConnection",
+				PrivateEndpointResourceId: "azure-pl-id",
+			},
+		}
+		connection.Status.Cloud = &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionStatusCloudOneOf{
+			NetworkingV1AzurePrivateLinkAttachmentConnectionStatus: &networkingprivatelinkv1.NetworkingV1AzurePrivateLinkAttachmentConnectionStatus{
+				Kind:                         "AzurePrivateLinkAttachmentConnectionStatus",
+				PrivateLinkServiceAlias:      "azure-pl-alias",
+				PrivateLinkServiceResourceId: "azure-pl-id",
```

Comment:
non blocking but it would be great if we can have real examples here and check how they render in the output